docs(*): Update for Explore Profiles (#160)
grafana/explore-profiles: docs/sources/determine-use-case/index.md

- Use case 2 - You want to research resource and performance hotspots to determine areas that can be optimized.

Your use case determines what’s most important.
For example, in use case 1, if a service is misbehaving, then you might want to see the profile types so you can see the CPU and memory profiles alongside each other.

For either use case, the first step is to identify areas of interest by reviewing profiles or a single service.
Selecting different profile types lets you focus on memory allocation, CPU processes, allocation sizes, blocks, or mutually exclusive (mutex).

The profile types available depend on how you've instrumented you app to generate your profiling data.
For more information, refer to [Profiling types](../concepts/#profile-types/) for help selecting a profile type to match your use case.
Refer to [Understand profile types](https://grafana.com/docs/pyroscope/latest/view-and-analyze-profile-data/profiling-types/) to learn about profile types and instrumentation methods.

After you’ve identified the problem process or service, you can filter and explore using labels and flame graphs to view lower levels.

> With capabilities like the Flame graph AI interpreter or the GitHub integration, Explore Profiles helps you locate the root cause and how to address it.

Author: Good catch. I'll add it.

> With capabilities like the Flame graph AI interpreter or the GitHub integration, Explore Profiles helps you locate the root cause and how to address it.

Author: Done.

## Use case 1: Investigate an issue

> Profiling data is idea when you know there is a specific service or area where there is a performance issue.

Reviewer: ```suggestion
Profiling data is ideal when you know there is a specific service or area where there is a performance issue.
```

Maybe an alert from spike in CPU led you to profiles or maybe your logs showed OOM issue for a particular service and you need to debug it.
Explore Profiles lets you quickly drill into a service and identify the performance issue in these scenarios.

### Example: Know the problem service, not the cause

> If you know the affected service (use case 1), your investigation starts by viewing that service using the **All services** view and reviewing the profiles for that service.

Reviewer: ```suggestion
If you know the affected service, your investigation starts by viewing that service using the **All services** view and reviewing the profiles for that service.
```

Author: I edited the suggestion to remove the specific example of using All services. Below, we have two sentences that provide methods for viewing the affected service.

> If you know the affected service (use case 1), your investigation starts by viewing that service using the **All services** view and reviewing the profiles for that service.

Author: I went back and forth on this. Which one would apply for more people in this situation? Is it better to point them to the Search service area and then go to Profile types or to start with Profile types and select a service from the drop-down? 

What if we add a sentence below to help explain.

> If you know the affected service (use case 1), your investigation starts by viewing that service using the **All services** view and reviewing the profiles for that service.

Author: I added another sentence below. See what you think.

> In the first scenario, you may have identified that the `checkoutservice` has an issue.

Author: ```suggestion
In this example, you may have identified that the `checkoutservice` has an issue.
```

> You can use **Search services** to locate `checkoutservice` and then use the **Profile types** view to examine all profiles for that service.

Author: ```suggestion
You can use **Search services** to locate `checkoutservice` and then use the **Profile types** view to examine all profiles for that service.

Alternatively, you can select the **Profile types** view and then choose the `checkoutservice` from the **Service** drop-down list.
```

### Example: Know there is an issue, need to investigate

> If you only know there is an issue and have to investigate, then your investigation starts by using the **All services** exploration type.

Reviewer: We should stick to "view" and not "exploration type" as the user will understand better IMO.

Using the **Profile type** selector, you can check the services’ CPU processes, memory allocation, blocks, locks, exceptions, and other available profile types.

After you locate the profile with a spike, select either **Profile types** view to examine all profile types for that service, or select **Labels** to view the labels (such as `hostname` or `span_name`) for that service.

## Use case 2: Research performance and resource bottlenecks

Illuminate performance issues across a number of dimensions when you are doing a proactive analysis.
This could be a cost-cutting exercise or attempt to improve latency or memory usage in a non-incident scenario. This exercise could be across multiple services or within a service across profile types, labels, or a combination of the two.

> The flow guides you through identifying the bottleneck.

Reviewer: Which flow?

Author: Let's remove the last sentence in that paragraph. We'll eventually write another use case to cover the cost-cutting scenario, but not right now. 
```suggestion
```
